src/LaunchDarkly.TestHelpers/HttpTest/RequestRecorder.cs

```diff
@@ -20,6 +20,8 @@ public class RequestRecorder
         public static readonly TimeSpan DefaultTimeout = TimeSpan.FromSeconds(5);
 
         private readonly BlockingCollection<RequestInfo> _requests = new BlockingCollection<RequestInfo>();
+        private readonly object _lock = new object();
+        private volatile bool _enabled = true;
```

Comment:
Both volatile and locking are going to create memory barriers. Is there some reason I am missing to use both?

Reply:
Nope - I realized that late & took out the lock before seeing your comment. A lock would make sense if there were a whole code path where the read or write of the field had to be done atomically with other things, but that's not the case here.